packets/play_clientbound.go

```diff
@@ -717,9 +719,8 @@ func (packet *SpawnPlayer) Parse(reader io.Reader) (err error) {
 	packet.Z, _ = ReadInt(reader)
 	packet.Yaw, _ = ReadByte(reader)
 	packet.Pitch, _ = ReadByte(reader)
-	packet.Metadata = NewEntityMetadata()
 
-	return packet.Metadata.Parse(reader)
+	return
```

Comment:
Nie może tak być, bo w ogóle nie sprawdzamy bledów podczas odczytu pakietu. Wcześniej było słabo napisane, ale ten Metadata.Parse by się wykrzaczył i zwrócił coś tam. W tej chwili nic nie zwracasz. Zrób `if err != nil` na każdym z tych ReadX powyżej.

Zasadniczo to powinien tutaj być jakiś error reader pattern w stylu bufio.Scanner na tych funkcjach od czytania typow z protokolu. https://dave.cheney.net/2019/01/27/eliminate-error-handling-by-eliminating-errors